### PR TITLE
Feat/list

### PR DESCRIPTION
**Pull request by Ebube111**

### CODE REVIEW

File: src/pages/global.css
Comment (carina-akaia) [inline src/pages/global.css:1]: I think this is not needed at all, radio and checkboxes should already use the primary color. Let me know if that's not the case ( and make sure you use UI kit components for checkboxes everywhere )

File: src/modules/lists/components/ListConfirmationModals.tsx [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+          >
+            View list
+          </button>
+        </div>
+      </div>
+    </div>
+  );
+};
+
+export default SuccessModalCreateList;
```
Comment (carina-akaia) [inline src/modules/lists/components/ListConfirmationModals.tsx:64]: please use named export

File: src/common/api/images.ts [hunk 2]
```diff
@@ -121,13 +121,13 @@ export const getImage = async ({
     } else if (socialImage?.ipfs_cid) {
       return `https://ipfs.near.social/ipfs/${socialImage.ipfs_cid}`;
     } else {
-      return fallbackurl ?? type === "image"
+      return (fallbackurl ?? type === "image")
```
[the remark this answers is not in the document]
Comment (Ebube111) [reply src/common/api/images.ts:124]: Done

File: src/common/api/images.ts [hunk 3]
```diff
@@ -121,13 +121,13 @@ export const getImage = async ({
...
         ? "/assets/images/profile-image.png"
         : "/assets/images/profile-banner.png";
     }
   } catch (err) {
     console.log("error fetching image ", err);
-    return fallbackurl ?? type === "image"
+    return (fallbackurl ?? type === "image")
```
[the remark this answers is not in the document]
Comment (Ebube111) [reply src/common/api/images.ts:130]: done

File: src/common/contracts/potlock/interfaces/lists.interfaces.ts [hunk 4]
```diff
@@ -37,3 +37,21 @@ export interface Registration {
 export interface GetListInput {
   list_id: string;
 }
+
+export interface ApplyToList {
+  list_id: string;
+  notes?: string;
```
[the remark this answers is not in the document]
Comment (Ebube111) [reply src/common/contracts/potlock/interfaces/lists.interfaces.ts:43]: done

File: src/common/assets/svgs/SuccessRedIcon.tsx
[the remark this answers is not in the document]
Comment (Ebube111) [reply src/common/assets/svgs/SuccessRedIcon.tsx:1]: done

File: src/modules/donation/components/DonationConfirmation.tsx [hunk 5]
```diff
@@ -1,3 +1,5 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+// @ts-nocheck
```
[the remark this answers is not in the document]
Comment (Ebube111) [reply src/modules/donation/components/DonationConfirmation.tsx:2]: cleared

File: src/common/contracts/potlock/lists.ts [hunk 6]
```diff
@@ -36,6 +99,97 @@ export const getList = (args: GetListInput) =>
     args,
   });
 
+export const registerBatch = (args: ApplyToList) =>
```
[the remark this answers is not in the document]
Comment (Ebube111) [reply src/common/contracts/potlock/lists.ts:106]: Done

File: src/modules/access-control/components/AccessControlListModal.tsx [hunk 7]
```diff
@@ -23,22 +24,50 @@ import {
...
+  countText?: string;
+  type?: "ADMIN" | "ACCOUNT";
+  contractAdmins?:
+    | { account: AccountId }[]
+    | { account: AccountId; id?: number }[];
 };
 
-export const AccessControlListModal = create(
-  ({ title, value: accountIds, onSubmit }: AccessControlListModalProps) => {
+export const AccessControlAccountsModal = create(
```
[the remark this answers is not in the document]
Comment (Ebube111) [reply src/modules/access-control/components/AccessControlListModal.tsx:42]: resolved

File: src/modules/access-control/components/AccessControlListModal.tsx [hunk 8]
```diff
@@ -23,22 +24,50 @@ import {
...
+    type,
+  }: AccessControlListModalProps) => {
     const self = useModal();
 
     const close = useCallback(() => {
       self.hide();
       self.remove();
     }, [self]);
 
+    const {
```
[the remark this answers is not in the document]
Comment (Ebube111) [reply src/modules/access-control/components/AccessControlListModal.tsx]: Done

File: src/modules/donation/hooks/fees.ts [hunk 9]
```diff
@@ -1,3 +1,5 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
```
[the remark this answers is not in the document]
Comment (Ebube111) [reply src/modules/donation/hooks/fees.ts:1]: Removed

File: src/modules/donation/models/index.ts [hunk 10]
```diff
@@ -1,3 +1,5 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
```
[the remark this answers is not in the document]
Comment (Ebube111) [reply src/modules/donation/models/index.ts:1]: removed

File: src/modules/lists/components/Donation/ConfirmDonation.tsx
[the remark this answers is not in the document]
Comment (Ebube111) [reply src/modules/lists/components/Donation/ConfirmDonation.tsx:1]: removed

File: src/modules/lists/components/DonationFlow.tsx
[the remark this answers is not in the document]
Comment (Ebube111) [reply src/modules/lists/components/DonationFlow.tsx:1]: moved

File: src/modules/lists/components/ListAccounts.tsx [hunk 11]
```diff
@@ -0,0 +1,174 @@
...
+            key={index}
+          />
+        ))}
+        {filteredRegistrations?.length === 0 && <p>No Registrations present</p>}
+      </div>
+    </div>
+  );
+};
+
+export default ListAccounts;
```
[the remark this answers is not in the document]
Comment (Ebube111) [reply src/modules/lists/components/ListAccounts.tsx:161]: resolved

File: src/modules/lists/components/ListDetails.tsx [hunk 12]
```diff
@@ -0,0 +1,324 @@
...
+
+  return (
+    <>
+      <div className="md:px-10">
+        <p className="mb-2 text-2xl font-semibold">{listDetails.name}</p>
+        <div className="mb-2 flex items-center space-x-2">
+          By{" "}
+          <img
+            className="ml-2 h-4 w-4 rounded-full object-cover"
+            src="https://images.unsplash.com/photo-1586297135537-94bc9ba060aa?w=800&auto=format&fit=crop&q=60&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8MTR8fHVzZXJ8ZW58MHx8MHx8fDA%3D"
```
[the remark this answers is not in the document]
Comment (Ebube111) [reply src/modules/lists/components/ListDetails.tsx]: resolved. it wasn't even fetching the owner's image. this was taken out

File: src/modules/lists/components/ListDetails.tsx [hunk 13]
```diff
@@ -0,0 +1,324 @@
...
+                ),
+            ),
+          )
+        }
+      />
+    </>
+  );
+};
+
+export default ListDetails;
```
[the remark this answers is not in the document]
Comment (Ebube111) [reply src/modules/lists/components/ListDetails.tsx:372]: resolved

File: src/modules/lists/hooks/useAllLists.ts [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
+      });
+      setRegistrations(upvotedLists);
+      setFilteredRegistrations(upvotedLists);
+      setCurrentListType("My Favorites");
+    } catch (error) {
+      console.error("Error fetching favorites:", error);
+    }
+  }, [wallet, setCurrentListType]);
+
+  const buttons = [
```
[the remark this answers is not in the document]
Comment (Ebube111) [reply src/modules/lists/hooks/useAllLists.ts:60]: resolved

File: src/modules/lists/hooks/useListForm.ts [hunk 15]
```diff
@@ -0,0 +1,189 @@
...
+  delete_list,
+  registerBatch,
+  remove_admins_from_list,
+  transfer_list_ownership,
+  unregister_from_list,
+} from "@/common/contracts/potlock/lists";
+import { AccountId } from "@/common/types";
+import { validateAccountId } from "@/modules/core";
+
+// Step 1: Define the enum
```
[the remark this answers is not in the document]
Comment (Ebube111) [reply src/modules/lists/hooks/useListForm.ts]: removed

File: src/pages/list/[id]/index.tsx [hunk 16]
```diff
@@ -0,0 +1,15 @@
+"use client";
+import React from "react";
+
+import ListAccounts from "@/modules/lists/components/ListAccounts";
+
+import { ListDetails } from "../../../modules/lists/components/ListDetails";
```
[the remark this answers is not in the document]
Comment (Ebube111) [reply src/pages/list/[id]/index.tsx:12]: resolved

File: src/pages/list/[id]/index.tsx [hunk 17]
```diff
@@ -0,0 +1,15 @@
+"use client";
+import React from "react";
+
+import ListAccounts from "@/modules/lists/components/ListAccounts";
```
[the remark this answers is not in the document]
Comment (Ebube111) [reply src/pages/list/[id]/index.tsx:8]: resolved

File: src/pages/list/[id]/index.tsx [hunk 18]
```diff
@@ -0,0 +1,15 @@
+"use client";
+import React from "react";
```
[the remark this answers is not in the document]
Comment (Ebube111) [reply src/pages/list/[id]/index.tsx]: removed

File: src/pages/global.css
Comment (Ebube111) [reply src/pages/global.css:1]: Yeah the custom checkbox in list details uses this style

File: src/modules/lists/components/ListConfirmationModals.tsx (hunk 1 above)
Comment (Ebube111) [reply src/modules/lists/components/ListConfirmationModals.tsx:67]: done

**Review by carina-akaia — APPROVED** [review]

**Pull request merged**